PackageViewModel/SignPackageViewModel.cs

```diff
@@ -0,0 +1,363 @@
+
+        private void ValidateCertificate()
+        {
+            if (_certificateValidationTimer == null)
+            {
+                _certificateValidationSemaphore = new SemaphoreSlim(1);
+                _certificateValidationTimer = new Timer(ValidateCertificateCallback);
+            }
+            _certificateValidationTimer.Change(250, Timeout.Infinite);
+        }
+
+        private void ValidateCertificateCallback(object state)
+        {
+            _certificateValidationSemaphore.Wait();
+
+            try
+            {
+                Clear();
+
+                X509Certificate2 certificate;
+
+                var certificateFileName = CertificateFileName;
+                if (string.IsNullOrEmpty(certificateFileName))
+                {
+                    certificate = Certificate;
+                }
+                else
+                {
+                    var password = Password;
+                    if (string.IsNullOrEmpty(password))
+                    {
+                        certificate = new X509Certificate2(certificateFileName);
+                    }
+                    else
+                    {
+                        // this throws if the password is wrong
+                        certificate = new X509Certificate2(certificateFileName, password);
+                    }
+                }
+
+                if (certificate == null)
+                {
+                    return;
+                }
+
+                using (var tempCertificate = new X509Certificate2(certificate))
+                // change to AuthorSignPackageRequest when NuGet.Client is updated
+                using (var signRequest = new SignPackageRequest(tempCertificate, GetHashAlgorithmName(tempCertificate)))
+                {
+                    SigningUtility.Verify(signRequest, NullLogger.Instance);
+                }
+
+                Certificate = certificate;
+                CanSign = true;
+            }
+            catch (System.Security.Cryptography.CryptographicException ex) when (ex.HResult == ERROR_INVALID_PASSWORD_HRESULT)
+            {
+                Certificate = null;
+
+                if (!ShowPassword)
+                {
+                    OnError(new Exception("Password required"));
+                    ShowPassword = true;
+                }
+                else
+                {
+                    OnError(new Exception("Invalid Password"));
+                }
+            }
+            catch (Exception ex)
+            {
+                Certificate = null;
+                OnError(ex);
+            }
+            finally
+            {
+                _certificateValidationSemaphore.Release();
+            }
+        }
+
+        // https://github.com/NuGet/NuGet.Client/blob/894388a598834a1fe8a0e483a2bc050c05693313/src/NuGet.Core/NuGet.Packaging/Signing/Utility/CertificateUtility.cs#L112-L124
+        private static HashAlgorithmName GetHashAlgorithmName(X509Certificate2 certificate)
```

Comment:
The hash algorithm chosen shouldn't have anything to do with the certificate. We should either hard-code Sha256 or put a drop-down dialog with the options and default to Sha256.

Reply:
Will do.